segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/DebugBookKeeperLogWrapper.java

```diff
@@ -301,11 +306,13 @@ public int deleteLedgersStartingWithId(long startId) throws DurableDataLogExcept
         ledgersToDelete.forEach(id -> {
             try {
                 Ledgers.delete(id, this.bkClient);
+                LOGGER.info("{}: Deleted ledger {}.", this.traceObjectId, id);
                 count.incrementAndGet();
             } catch (DurableDataLogException ex) {
-                ex.printStackTrace();
+                LOGGER.error("{}: Unable to delete ledger {}.", this.traceObjectId, id, ex);
             }
         });
+        forceMetadataOverWrite(metadata);
```

Comment:
This metadata is outdated, right? I think that what we need to do after this is a reconcile. In my understanding, this is just rewriting the same metadata again. Please, do some tests to verify that.

Reply:
removed this line. Will be adding docs for recovery process to include reconcile as next step.